### ISSUE TITLE
ActiveRecord silently ignores the query on RangeError when ".any?" or ".exists?" is being used

### ISSUE THREAD

**Issue by kpumuk** [rails/rails#32609]
### Steps to reproduce

```
ruby-2.4.2 >> User.where(id: 123984738957498574).any?
→ false

ruby-2.4.2 >> User.where(id: 123984738957498574).exists?
→ false

ruby-2.4.2 >> User.where(id: 123984738957498574, non_existing_column: 'really?').any?
→ false

ruby-2.4.2 >> User.where(id: 123984738957498574).to_sql
ActiveModel::RangeError: 123984738957498574 is out of range for ActiveModel::Type::Integer with limit 4 bytes

ruby-2.4.2 >> User.where(id: 1).any?
  User Exists (1.0ms)  SELECT  1 AS one FROM `users` WHERE `users`.`id` = 1 LIMIT 1
→ true
```

### Expected behavior

It should raise an `ActiveModel::RangeError` error.

### Actual behavior

Query is being silently ignored.

### System configuration
**Rails version**: 5.1.5, 5.2.9

**Ruby version**: 2.4.2, 2.5.1

**Comment by y-yagi** [comment]
This is intentional behavior. 
`.any?` and `.exists?"` are designed to return boolean values. 
Ref: #25271

**Comment by kpumuk** [comment]
Another example:

```
ruby-2.4.2 >> User.where.not(id: 34749857498754985794).exists?
→ false
```

**Comment by kamipo** [comment]
This behavior will be fixed by #30000.

**Comment by kpumuk** [comment]
Thank you
